pkg/api/defaults.go

```diff
@@ -404,7 +404,7 @@ func (p *Properties) setExtensionDefaults() {
 }
 
 func (p *Properties) setMasterProfileDefaults(isUpgrade, isScale bool, cloudName string) {
-	if p.MasterProfile.Distro == "" {
+	if p.MasterProfile.Distro == "" && p.MasterProfile.ImageRef == nil {
```

Comment:
maybe we should assign a Distro value `custom` or something like that so it's explicit and doesn't run into unexpected usages of empty distro value.

Reply:
I have a preference for using empty string as a more expressive/semantic value of "this cluster is not using any distro configuration"

Reply:
@mboersma @devigned @PatrickLang @marosset any thoughts here?

Reply:
I think setting to `nil` is most clear if we want to say "no distro set." But that may have unfortunate ripple effects in the code, so I'm fine with empty string.

Reply:
Hmm... might be taste, but I'd vote for either `*string` to indicate emptiness or just document what `""` means. I worry that adding a default that is not a golang idiomatic default value would further complicate. If we were to move down that path, then I think I'd not use a string, but rather a type (an enumeration).

Reply:
I don't understand this part

> adding a default that is not a golang idiomatic default value would further complicate

Is it non-idiomatic to assign significance to the zero value of a type? (in this case a string, so `""`)

Reply:
I'm sorry for not being clear. `""` is the idiomatic default value for string. In contrast, a default of `"custom"` would not. If we were to use `"custom"`, I would vote we use a named type for `Distro` rather than a string.